Enable custom aggregate functions (take 2) (#529)
sql-js/sql.js: README.md

// Run a query in which the function is used
db.run("INSERT INTO hello VALUES (add_js(7, 3), add_js('Hello ', 'world'));"); // Inserts 10 and 'Hello world'

// You can create custom aggregation functions, by passing a name
// and a set of functions to `db.create_aggregate`:
//

> // - a initialization function. This function receives no arguments and returns

Reviewer: ```suggestion
// - an initialization function. This function receives no arguments and returns
```

> // - an initialization function. This function receives no arguments and returns

Reviewer: ```suggestion
// - an initialization function. This function receives no argument and returns
```

> // - an initialization function. This function receives no argument and returns
> //   the initial value for the aggregation function
> // - a step function. This function receives as a first argument the state
> //   object created in init, as well as the values received in the step. It
> //   will be called on every value to be aggregated, and its return value
> //   will be used as the state for the next iteration.
> // - a finalizer. This function receives one argument, the state object, and
> //   returns the final value of the aggregate. It can be omitted, in which case
> //   the value of the `state` variable will be used.

Reviewer: ```suggestion
// - an `init` function. This function receives no argument and returns
//   the initial value for the state of the aggregate function.
// - a `step` function. This function takes two arguments
//    - the current state of the aggregation
//    - a new value to aggregate to the state
//  It should return a new value for the state.
// - a `finalize` function. This function receives a state object, and
//   returns the final value of the aggregate. It can be omitted, in which case
//   the final value of the state will be returned directly by the aggregate function.
```

//
// Here is an example aggregation function, `json_agg`, which will collect all
// input values and return them as a JSON array:
db.create_aggregate(
  "json_agg",
  {
    init: () => [],
    step: (state, val) => [...state, val],
    finalize: (state) => JSON.stringify(state),
  }
);

> ```suggestion

Reviewer: ```suggestion
```

> // Now if you run this query:
> var result = db.exec("SELECT json_agg(somecol) FROM atable;");
> console.log("You'll get a json-encoded list of values: ", result[0].values[0])

Reviewer: ```suggestion
```suggestion
db.exec("SELECT json_agg(column1) FROM (VALUES ('hello'), ('world'))");
// -> The result of the query is the string '["hello","world"]'
```
```

// Run a query in which the function is used
db.run("INSERT INTO hello VALUES (add_js(7, 3), add_js('Hello ', 'world'));"); // Inserts 10 and 'Hello world'

// You can create custom aggregation functions, by passing a name
// and a set of functions to `db.create_aggregate`:
//
// - a initialization function. This function receives no arguments and returns
//   the initial value for the aggregation function
// - a step function. This function receives as a first argument the state
//   object created in init, as well as the values received in the step. It
//   will be called on every value to be aggregated, and its return value
//   will be used as the state for the next iteration.
// - a finalizer. This function receives one argument, the state object, and
//   returns the final value of the aggregate. It can be omitted, in which case
//   the value of the `state` variable will be used.
//
// Here is an example aggregation function, `json_agg`, which will collect all
// input values and return them as a JSON array:
db.create_aggregate(
  "json_agg",
  {
    init: () => [],

>     step: (state, val) => state.push(val),

Reviewer: This doesn't work anymore, does it ?

```suggestion
    step: (state, val) => [...state, val],
```

Author: yup, good catch
